### PR TITLE
chore(e2e): Update test attributes for new builds

### PR DESCRIPTION
**Pull request by joshbrand**
This is mostly scenario support for future pipeline work and some housekeeping cleanup. All new attributes have safe defaults won't break current tests. Also, updates the worker module to use new provider attribute for ipv4 public addressing and simplifes variables in the workflow.

### CODE REVIEW

File: .github/workflows/enos-run.yml [hunk 1]
```diff
@@ -80,7 +83,19 @@ jobs:
...
+      ENOS_VAR_aws_ssh_keypair_name: ${{ github.event.repository.name }}-ci-ssh-key
+      ENOS_VAR_aws_ssh_private_key_path: ./support/private_key.pem
+      ENOS_VAR_local_boundary_dir: ./support/boundary
+      ENOS_VAR_local_boundary_ui_dir: ./support/boundary-ui
+      ENOS_VAR_crt_bundle_path: ./support/boundary.zip
+      ENOS_VAR_tfc_api_token: ${{ secrets.TF_API_TOKEN }}
+      ENOS_VAR_test_email: ${{ secrets.SERVICE_USER_EMAIL }}
+      ENOS_VAR_boundary_edition: ${{ inputs.edition }}
+      ENOS_VAR_boundary_docker_image_name: ${{ inputs.docker-image-name }}
+      ENOS_VAR_boundary_docker_image_file: ./support/boundary_docker_image.tar
```
Comment (moduli) [inline .github/workflows/enos-run.yml:98]: Nice

File: enos/modules/read_license/main.tf [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# Copyright (c) HashiCorp, Inc.
+# SPDX-License-Identifier: MPL-2.0
+
+variable "file_name" {}
+
+output "license" {
+  value = file(var.file_name)
```
Comment (moduli) [inline enos/modules/read_license/main.tf:7]: How does this behave if the file_name is invalid/non-existent? It seems like the runs on this PR worked fine, but I was a little surprised by that.

File: enos/enos-scenario-e2e-static-with-vault.hcl [hunk 3]
```diff
@@ -16,6 +16,7 @@ scenario "e2e_static_with_vault" {
   locals {
     aws_ssh_private_key_path = abspath(var.aws_ssh_private_key_path)
     boundary_install_dir     = abspath(var.boundary_install_dir)
+    boundary_license_path    = abspath(var.boundary_license_path != null ? var.boundary_license_path : joinpath(path.root, "./support/boundary.hclic"))
```
Comment (moduli) [inline enos/enos-scenario-e2e-static-with-vault.hcl:19]: I was thinking about the `joinpath(path.root, "./support/boundary.hclic")` part here and was wondering if that was the appropriate thing to set here. I haven't thought through alternatives, though...

Comment (joshbrand) [reply enos/enos-scenario-e2e-static-with-vault.hcl:19]: it's a default we use in vault for CI, but up for changes!

Comment (moduli) [reply enos/enos-scenario-e2e-static-with-vault.hcl:19]: We can just leave it like this for now.

File: enos/modules/read_license/main.tf (hunk 2 above)
Comment (joshbrand) [reply enos/modules/read_license/main.tf:7]: it bails out quickly when planning the scenario:
```Error: Invalid function argument

  on ../../modules/read_license/main.tf line 4, in output "license":
   4:   value = file(var.file_name)
    ├────────────────
    │ while calling file(path)
    │ var.file_name is "/tmp/fakenotafile.hclic"
```

Comment (moduli) [reply enos/modules/read_license/main.tf:7]: So this needs to be set in oss? EDIT: Just kidding. I missed the `skip_step` in the read license step.

**Review by moduli — APPROVED** [review]

**Pull request merged**